File: Rediscuss.Microservices/services/Rediscuss.ForumService/DTOs/CreatePostDto.cs
﻿using System.ComponentModel.DataAnnotations;

namespace Rediscuss.ForumService.DTOs
{
    public class CreatePostDto
    {
        [Required]
        public string Title { get; set; }
        public string? Content { get; set; }

        [Required]
        public string DiscuitId { get; set; }
    }
}

File: Rediscuss.Microservices/services/Rediscuss.ForumService/src/posts/dto/create-post.dto.ts
import { ApiProperty } from '@nestjs/swagger';

export class CreatePostDto{
    @ApiProperty()
    title: string;

    @ApiProperty()
    content: string;

    @ApiProperty({ description: 'Discuit id for the post' })
    discuitId: string;

    @ApiProperty({ description: 'Topic ids for the post', type: [String], required: false })
    topicIds?: string[];
}